xl-objects-multipart.go

```diff
@@ -222,22 +220,20 @@ func (xl xlObjects) CompleteMultipartUpload(bucket string, object string, upload
 		return "", toObjectErr(err, bucket, multipartObjFile)
 	}
 
-	// Attempt a rename of the upload id to temporary location, if
-	// successful then delete it.
+	// Attempt a rename of the upload id to temporary location, if successful then delete it.
 	uploadIDPath := path.Join(mpartMetaPrefix, bucket, object, uploadID, incompleteFile)
 	tempUploadIDPath := path.Join(tmpMetaPrefix, bucket, object, uploadID, incompleteFile)
 	if err = xl.storage.RenameFile(minioMetaBucket, uploadIDPath, minioMetaBucket, tempUploadIDPath); err == nil {
 		if err = xl.storage.DeleteFile(minioMetaBucket, tempUploadIDPath); err != nil {
 			return "", toObjectErr(err, minioMetaBucket, tempUploadIDPath)
 		}
-		return s3MD5, nil
-	}
-	// Rename if failed attempt to delete the original file.
-	err = xl.storage.DeleteFile(minioMetaBucket, uploadIDPath)
-	if err != nil {
-		return "", toObjectErr(err, minioMetaBucket, uploadIDPath)
+	} else {
+		// If Rename failed control reaches here, delete the original file.
```

Comment:
If rename from orig->tmp failed we are supposed to rename it back tmp->orig right?